tests/infrastructure/virt-controller-leader-election.go

```diff
@@ -55,12 +50,9 @@ var _ = DescribeInfra("Start a VirtualMachineInstance", func() {
 			Eventually(libinfra.GetLeader, 30*time.Second, 5*time.Second).ShouldNot(Equal(leaderPodName))
 
 			By("Starting a new VirtualMachineInstance")
-			vmi := libvmifact.NewAlpine()
-			obj, err := virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(testsuite.GetTestNamespace(vmi)).Body(vmi).Do(context.Background()).Get()
+			vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), libvmifact.NewGuestless(), metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			vmiObj, ok := obj.(*v1.VirtualMachineInstance)
-			Expect(ok).To(BeTrue(), "Object is not of type *v1.VirtualMachineInstance")
-			libwait.WaitForSuccessfulVMIStart(vmiObj)
+			libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Pending})
```

Comment:
This comes down to 
```
}, time.Duration(w.timeout)*time.Second, 1*time.Second).Should(gomega.BeElementOf(w.phases),
```
So we really need to include all possible phases otherwise this is flake.

Reply:
> So we really need to include all possible phases otherwise this is flake.

Can I do a full circle  and return to the original `WaitForSuccessfulVMIStart` ? I don't think it makes sense for this test to mention all (current, future) intermediately phases.

Reply:
>Can I do a full circle and return to the original WaitForSuccessfulVMIStart ?
That would mean you close the PR or not?

> don't think it makes sense for this test to mention all (current, future) intermediately phases.
A phase cannot be added per se.

What we ideally want is to say the phase should not be unset.

Eventually(matcher.ThisVMI(vmi)).WithTimeout(time.Minute).WithPolling(time.Second).Should(Not(matcher.BeInPhase("")))

Reply:
Sorry for not being clear. 
The other phases are required as the `WaitForVMIPhase` is checking the phase in polling and not in watch. So the phase `Pending` can be missed as the VMI might have move to the next phase really fast. That would mean that the test becomes flaky. Therefore we need to provide also phases after `Pending`. 
I hope it is more clear now

Reply:
> I hope it is more clear now

It does not. As it does not refer to my question

> Can I do a full circle and return to the original WaitForSuccessfulVMIStart ?

Reply:
>    Can I do a full circle and return to the original WaitForSuccessfulVMIStart ?
You are free to do so but it would be shame not to make the test slightly faster.

Reply:
> it would be shame not to make the test slightly faster.

This PR is not perfect, but improves readability and reduced time and memory requirements by using Guestless. As a net positive to code quality, I hope you approve it as it is now.

Reply:
This is a serial test; if the difference between a scheduled/running (guestless) VM is significant on an (essentially empty) cluster, then we're all in huge trouble with total test runtime, and thus, to me, it's not worth the micro optimization

Reply:
Thank you for this discussion.
While I see @xpivarc's point I think we agree it's not a hard blocker for this PR, since it is an overall improvement. We can re-iterate it in a follow-up.

Reply:
> This is a serial test; if the difference between a scheduled/running (guestless) VM is significant on an (essentially empty) cluster, then we're all in huge trouble with total test runtime, and thus, to me, it's not worth the micro optimization

Right, optimizing the time it takes to execute test is much more valuable than the resources, in case of Serial tests. No objection to merge this PR